wire-library/wire-protoc/src/main/java/com/squareup/wire/protocwire/WireGenerator.kt

```diff
@@ -110,18 +112,18 @@ class WireGenerator(
   }
 }
 
-fun parseFileDescriptor(fileDescriptor: DescriptorProtos.FileDescriptorProto): ProtoFileElement {
+fun parseFileDescriptor(fileDescriptor: DescriptorProtos.FileDescriptorProto, descs: DescriptorSource): ProtoFileElement {
   val location = Location.get(fileDescriptor.name)
   val imports = mutableListOf<String>()
   val publicImports = mutableListOf<String>()
-  val types = mutableListOf<MessageElement>()
-  val nestedTypes = mutableListOf<MessageElement>()
+  val types = mutableListOf<TypeElement>()
   for (messageType in fileDescriptor.messageTypeList) {
-    for (nestedType in messageType.nestedTypeList) {
-      nestedTypes.add(parseMessage(nestedType))
-    }
```

Comment:
I noticed this unused variable `nestedTypes` and also realized that this is not the way to handle nested types: they must be handled recursively since messages can be nested arbitrarily deeply.